lib/fontist/indexes/base_index.rb

```diff
@@ -0,0 +1,82 @@
+require_relative "index_formula"
+
+module Fontist
+  module Indexes
```

Comment:
I'm a bit confused -- what does "Indexes" mean here? Different types of indexes?

Reply:
There are two indexes now: `FontIndex` and `FilenameIndex`. The first one searches by font name, and the second one by filename.

It seemed to me easier to implement filename search by reusing `FontIndex`.

The name `Indexes` confusing me a little too: class names are too long.

Any thoughts on improvement?

Reply:
In this case, we should apply the modeling principles — if you have a Truck and a Bus, they both subclass from Vehicle, not Vehicles. So FileIndex and FontIndex should both inherit from Index, not Indexes 😉

Reply:
Actually, scrap that. Let’s go with Indexes since it’s a Module.